src/Command/AnalyzeCommand.php

```diff
@@ -55,23 +55,24 @@ public function __construct(
     protected function configure()
     {
         $this->setName('analyze');
-
-        $this->getDefinition()->setArguments([
-            new InputArgument('depfile', InputArgument::OPTIONAL, 'Path to the depfile', getcwd().'/depfile.yml'),
-        ]);
+        $this->addArgument(
+            'depfile',
+            InputArgument::OPTIONAL,
+            'Path to the depfile',
+            getcwd().'/depfile.yml'
+        );
 
         $this->getDefinition()->addOptions($this->formatterFactory->getFormatterOptions());
     }
 
-    protected function execute(
-        InputInterface $input,
-        OutputInterface $output
-    ) {
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
         ini_set('memory_limit', -1);
 
         $this->printBanner($output);
 
-        $configurationLoader = new ConfigurationLoader($input->getArgument('depfile'));
+        $depfile = $input->getArgument('depfile');
```

Comment:
no need to declare this variable?

Reply:
At this moment not, but `ConfigurationLoader` is too stateful and should be refactored